File: POJ/1852/Main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var t int
	fmt.Scan(&t)
	for i := 0; i < t; i++ {
		var l float64
		var n int
		fmt.Scan(&l)
		fmt.Scan(&n)
		var max, min float64 = 0, 0
		for j := 0; j < n; j++ {
			var x, m float64
			fmt.Scan(&x)
			m = math.Min(l - x, x)
			min = math.Max(min, m)
			m = math.Max(l - x, x)
			max = math.Max(max, m)
		}
		fmt.Printf("%d %d", int(min), int(max))
	}
}

File: POJ/1852/Main.java
import java.util.Scanner;

public class Main {
    void run() {
        Scanner scan = new Scanner(System.in);
        int t = scan.nextInt();
        for (int i = 0; i < t; i++) {
            int l = scan.nextInt();
            int n = scan.nextInt();
            int min = 0, max = 0;
            for (int j = 0; j < n; j++) {
                int x = scan.nextInt();
                int m;
                m = Math.min(l - x, x);
                min = Math.max(min, m);
                m = Math.max(l - x, x);
                max = Math.max(max, m);
            }
            System.out.println(min + " " + max);
        }
    }
    public static void main(String[] args) {
        new Main().run();
    }
}
